src/Models/Traits/HasCustomAttributes.php

```diff
@@ -79,9 +80,13 @@ public function scopeAttributeHas(Builder $builder, string $attributeCode, calla
 
     public function scopeWhereAttribute(Builder $builder, string $attributeCode, $operator = null, $value = null)
     {
+        $attributeId = EavAttribute::getAttributeId($attributeCode);
+
         return $builder->attributeHas(
             $attributeCode,
-            fn ($query) => $query->where('value', $operator, $value)->where($this->getCustomAttributeCode(), $attributeCode)
+            fn ($query) => $query
+                ->where('value', $operator, $value)
+                ->where('attribute_id', $attributeId)
```

Comment:
`$this->getCustomAttributeCode()` was used so that we can add CustomAttributes to more than just products. For example, `ProductLink` has the custom `product_link_attribute_code` attribute. With this that would currently be broken.